Add support for linux-gcc-aarch64.

david-allison opened the pull request:
> [!WARNING]
> I have never written a Makefile before and have limited knowledge of compiler flags.

The main purpose of this PR is to add support for `linux-gcc-aarch64`.

This patch enables me to run `ufal/udpipe` under Docker on my M1 MacBook.

I also cleaned up CI ~~and copyright headers~~. Take or leave these changes

### Testing

* I have fixed CI and added Arm runners.
  * I have confirmed the [newly added Arm runners failed before the change](https://github.com/david-allison/cpp_builtem/actions/runs/14740838841/job/41378163654)
* Using [`ufal/udpipe`](https://github.com/ufal/udpipe/tree/master) I loaded a custom model and successfully ran:
  * `docker build --no-cache -t udpipe/server .`
  * `docker run --rm -it -p 8080:8080 udpipe/server` 
  * `curl -s -F data=@data.txt -F tokenizer= -F tagger= -F parser= http://127.0.0.1:8080/process | PYTHONIOENCODING=utf-8 python3 -c "import sys,json; sys.stdout.write(json.load(sys.stdin)['result'])"`

david-allison committed:
Add support for linux-gcc-aarch64.

`$(uname -m)` returns `aarch64` on `ubuntu-24.04-arm` (GitHub Actions)

Fixes `udpipe` Docker build on an M1 Mac

david-allison committed:
Update manylinux to glibc 2.28.

Partially reverts 2b916627d98d927a2338f3a47c620e2f202de8be

Node.js 20 requires at least glibc 2.28

Fixes:

```
/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /__e/node20/bin/node)
```

david-allison commented on Makefile.builtem:
```diff
@@ -63,7 +65,7 @@ else ifneq ($(filter macos-clang,$(PLATFORM)),)
 
 # fail for unknown platforms
-PLATFORMS = linux-gcc-64 linux-gcc-32 linux-clang-64 linux-clang-32
+PLATFORMS = linux-gcc-64 linux-gcc-32 linux-gcc-aarch64 linux-clang-64 linux-clang-32
```
Note that this is inconsistent: `macos` uses `macos-clang-arm64`, and linux uses `linux-gcc-aarch64`.

Rust toolchains use `aarch64` for both, flagging this as you may wish me to change this.

foxik commented:
Thanks a lot for your work! I am generally open to adding it. However, I am currently swamped by the finishing semester, so I have taken a break from engineering work for a while. Therefore, I will review your addition after the end of the semester; most likely at the beginning of June. Cheers!

david-allison commented:
@foxik No rush on this. Thanks for the response and I'll aim to follow up in June. 

Thank you for UDPipe!

david-allison commented:
@foxik Following up (no pressure at all, just going through my pending PRs)

david-allison commented:
* rebased on `master`. I fixed the build in this PR, and similar fixes were applied and conflicted
  * My changes fixed the build, vs removing it in 2b916627d98d927a2338f3a47c620e2f202de8be, so I retained this commit
* Dropped copyright header commit: https://github.com/ufal/cpp_builtem/commit/a59e6245c4e3791105a53b7337fb51e92cc74f64
  * Makes for an easier review

unhammer commented:
Why was this closed?

david-allison commented:
There was no reply for over half a year and it didn't seem worthwhile chasing/rebasing. 

Feel free to take this through to completion if it's useful for others.

foxik commented:
Yeah, that is totally my fault. Sorry for taking too long. It is not that I am against it, but I ended up pursuing other priorities at work, and at some point @david-allison gave up (which in some sense is the best possible reaction).

I am not sure if this is still useful for you @david-allison, but I will still try to boost this up in my priorities.

david-allison commented:
@foxik No worries! Been there myself.

I don't need this in as I'm happy using my fork, but I suspect @unhammer would like to see this `master` for use in nix:

* https://github.com/NixOS/nixpkgs/pull/458819#issuecomment-3987008967

I don't plan to see this through now, but I'm happy for anyone to build on my work [or replace it]. 

Explicitly: all commits for this PR are under the Mozilla Public License Version 2.0
